### PR TITLE
Overhaul of the Python API

### PR DESCRIPTION
**Pull request by nagitsu**
Adds several utility functions to ease the use of Luminoth through Python:
* `Detector` (in `luminoth.tasks`) to abstract the usage of object detection.
* `vis_objects` to visualize the results by drawing the bounding boxes in the image.
* `read_image` to read images from disk without needing to know about Pillow.

With this, the following is now possible:
```python
from luminoth import Detector, read_image, vis_objects

image = read_image('cat.png')

# Assumes `accurate` checkpoint by default. Other checkpoint, or a config object,
# can be specified.           
detector = Detector()
objects = detector.predict(image)

print(objects)

vis_objects(image, objects).save('cat-out.png')
```

Further work is required along the following lines:
* I/O functions to read batchs of frames out of a video.
* Remove the TODOs in `Detector` that depend on rewriting `PredictorNetwork`.
* Provide a way to access the detected objects' embeddings for further processing.
* Document the new API.
* Add examples on processing both images and videos.

### CODE REVIEW

File: luminoth/tasks.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    In order to perform object detection with a model implemented within
+    Luminoth, this class should be used.
+
+    Attributes:
+        classes (list of str): Ordered class names for the detector.
+        prob (float): Default probability threshold for predictions.
+
+    TODO:
+        - Don't create a TF session internally (or make its creation optional)
```
Comment (dekked) [inline luminoth/tasks.py]: Could we re-use existing session if Luminoth is run via Python/Notebook?

File: luminoth/tasks.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+          the graph and makes the checkpoint loading fail).
+    """
+
+    DEFAULT_CHECKPOINT = 'accurate'
+
+    def __init__(self, checkpoint=None, config=None, prob=0.7, classes=None):
+        """Instantiate a detector object with the appropriate config.
+
+        Arguments:
+            checkpoint (str): Checkpoint name to instantiate the detector as.
```
Comment (dekked) [inline luminoth/tasks.py]: Should say that `checkpoint` can actually be an id or alias.

File: luminoth/tasks.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+        if prob is None:
+            prob = self.prob
+
+        if classes is None:
+            classes = self.classes
+        else:
+            classes = set(classes)
+
+        # TODO: Remove the loop. Neither Faster R-CNN nor SSD support batch
```
Comment (dekked) [inline luminoth/tasks.py]:
```
# TODO: Remove the loop once (and if) we implement batch sizes.
```

Make it not seem like it's trivial work :joy:

File: luminoth/vis.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    if colormap is None:
+        colormap = build_colormap()
+
+    image = Image.fromarray(image.astype(np.uint8))
+
+    draw = ImageDraw.Draw(image, 'RGBA')
+    for obj in objects:
+        # TODO: Can we do image resolution-agnostic?
+        color = colormap(obj['label'])
+        draw_rectangle(draw, obj['bbox'], color, width=3)
```
Comment (dekked) [inline luminoth/vis.py]: It would be nice to have `width` (and `fill`) as options.

File: luminoth/tasks.py (hunk 1 above)
Comment (nagitsu) [reply luminoth/tasks.py]: For notebooks, the problem is mainly the second point: internally `PredictorNetwork` always recreates the model, thus we end up with name clashes and suffixes. We must discard the old graph or find a way around that.

I haven't checked if there's a way to get the session from within. But in the case of TF-Eager, we simply don't want a session to exist (we'll probably have to check with `tf.executing_eagerly`).

File: luminoth/vis.py (hunk 4 above)
Comment (nagitsu) [reply luminoth/vis.py]: Will add generic `scale` parameter which modifies the font size too, or sizes will be inconsistent.

**Pull request merged**